core/src/test/java/org/elasticsearch/explain/ExplainActionTests.java

```diff
@@ -138,7 +139,7 @@ public void testExplainWithFields() throws Exception {
         assertThat(response.getExplanation().getValue(), equalTo(1.0f));
         assertThat(response.getGetResult().isExists(), equalTo(true));
         assertThat(response.getGetResult().getId(), equalTo("1"));
-        assertThat(response.getGetResult().getFields().size(), equalTo(1));
+        assertThat(response.getGetResult().getFields().size(), greaterThanOrEqualTo(1));
```

Comment:
ok, I didn't know that, thanks. I wonder if there is some way we can get the list of fields that have been randomly added? The problem I see here is that we aren't really testing that only the requesting fields are returned.